dotCMS/src/main/java/com/dotcms/cms/login/LoginServiceAPIImpl.java

```diff
@@ -0,0 +1,537 @@
+
+
+            HttpSession ses = request.getSession();
+
+
+            ses.setAttribute(com.dotmarketing.util.WebKeys.CMS_USER, user);
+            ses.setAttribute(WebKeys.USER_ID, user.getUserId());
+            SecurityLogger.logInfo(this.getClass(), "User " + userIdOrEmail
+                            + " has sucessfully login from IP: " + request.getRemoteAddr());
+
+
+            return true;
+
+        } catch (NoSuchUserException e) {
+            Logger.error(this.getClass(), "User " + userIdOrEmail + " does not exist.", e);
+            SecurityLogger.logInfo(this.getClass(), "An invalid attempt to login as "
+                            + userIdOrEmail + " has been made from IP: " + request.getRemoteAddr());
+            throw e;
+
+        } catch (Throwable e) {
+
+
+
+            if (user != null) {
+                user.setFailedLoginAttempts(user.getFailedLoginAttempts() + 1);
+                try {
+                    APILocator.getUserAPI().save(user, systemUser, false);
+                } catch (DuplicateUserException | DotDataException | DotSecurityException e1) {
+                    Logger.error(this.getClass(), "Login Failed: " + e1);
+                }
+            }
+            SecurityLogger.logInfo(this.getClass(), "An invalid attempt to login as "
+                            + userIdOrEmail + " from IP: " + request.getRemoteAddr());
+
+
+            Logger.error(this.getClass(), "Login Failed: " + e);
+            SecurityLogger.logInfo(this.getClass(), "An invalid attempt to login as "
+                            + userIdOrEmail + " has been made from IP: " + request.getRemoteAddr());
+        }
+
+        return false;
+    }
+
+    @Override
+    public boolean doLogin(final String userName, final String password)
+                    throws NoSuchUserException {
+
+        HttpServletRequest request =
+                        new MockSessionRequest(new MockHttpRequest("fake", "fake").request())
+                                        .request();
+        HttpServletResponse response =
+                        new MockHttpResponse(new BaseResponse().response()).response();
+
+        return _doLogin(userName, password, false, request, response, false);
+
+
+
+    }
+
+    @Override
+    public boolean passwordMatch(String password, User user) throws DotSecurityException {
+
+
+        try {
+            boolean needsToRehashPassword = false;
+            if (PasswordFactoryProxy.isUnsecurePasswordHash(user.getPassword())) {
+                // This is the legacy functionality that we will removed with
+                // the new hash library
+                boolean match = user.getPassword().equals(password) || user.getPassword()
+                                .equals(PublicEncryptionFactory.digestString(password));
+
+                // Bad credentials
+                if (!match) {
+                    throw new DotSecurityException("passwords don't match");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "passwords don't match" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)